js_modules/dagster-ui/packages/ui-core/src/assets/insights/AssetCatalogRateCard.tsx

```diff
@@ -2,26 +2,80 @@ import {BodyLarge, Box, Colors, Icon, Spinner} from '@dagster-io/ui-components';
 import React from 'react';
 
 import styles from './AssetCatalogRateCard.module.css';
-import {numberFormatter} from '../../ui/formatters';
+import {numberFormatter, percentFormatter} from '../../ui/formatters';
 
 export interface AssetCatalogRateCardProps {
   title: string;
   value: number | null; // e.g., 0.981 for 98.1%
   prevValue: number | null; // e.g., 1 for 100%
   loading: boolean;
+  unit: 'percent' | 'seconds';
+}
+
+function pctChange(x: number, y: number): number {
+  if (x === 0) {
+    if (y === 0) {
+      return 0.0; // No change
+    }
+    return Number.POSITIVE_INFINITY;
+  }
+
+  if (y > x) {
+    return y / x - 1;
+  }
+  return -(1 - y / x);
+}
+
+function formatValues(
+  valueOrNull: number | null,
+  prevValueOrNull: number | null,
+  unit: 'percent' | 'seconds',
+): {
+  currValueString: string;
+  prevValueString: string;
+  absDeltaString: string;
+  hasNegativeDelta: boolean;
+} {
+  const value = valueOrNull ?? 0;
+  const prevValue = prevValueOrNull ?? 0;
+
+  const delta = pctChange(prevValue, value);
+  const hasNegativeDelta = delta < 0;
+  const absDelta = Math.abs(delta);
+
+  if (unit === 'percent') {
+    return {
+      currValueString: percentFormatter.format(value),
+      prevValueString: percentFormatter.format(prevValue),
+      absDeltaString: percentFormatter.format(absDelta),
+      hasNegativeDelta,
+    };
+  }
+
+  const secondsFormatter = new Intl.NumberFormat(navigator.language, {
+    style: "unit",
+    unit: "second",
+  })
```

Comment:
This can be defined as a module-level const, since it's going to be an identical `NumberFormat` object for every call of `formatValues()`.

Reply:
good call